src/validateWebSite.ts

```diff
@@ -88,6 +88,10 @@ export async function validateWebSite(deploymentCorrelationId: string, siteTreeI
             statusCodes.push({ code: currentStatusCode, elapsed: elapsedSeconds });
 
             if (Date.now() > start + maximumValidationMs) {
+
+                if (currentStatusCode !== undefined && currentStatusCode >= 300) {
```

Comment:
`throw new Error();` is too generic. You don't want to use detectors for _any_ error, just status code errors. Maybe just change `validateWebSite` to return the status code.